seed/seedwriter/writer.go

```diff
@@ -655,13 +715,14 @@ func (w *Writer) InfoDerived() error {
 
 // SetInfo sets Info of the SeedSnap and possibly computes its
 // destination Path.
-func (w *Writer) SetInfo(sn *SeedSnap, info *snap.Info) error {
+func (w *Writer) SetInfo(sn *SeedSnap, info *snap.Info, seedComps []SeedComponent) error {
```

Comment:
This is maybe a little strange? Passing in a param to just set a field on the `SeedSnap`.

Reply:
In the end I decided to add a new method, `Writer.AddComponentsToSnap` to avoid mixing concerns with `SetInfo`. Although it just sets a variable.